pkg/openai/openai.go

```diff
@@ -29,12 +28,25 @@ func NewClient(apiSecret string, apiHost string) *Client {
 
 // truncateContentBasedOnTokens 基于 token 计算的方式截断文本
 func (c *Client) TruncateContentBasedOnTokens(textContent string, limits int) (string, error) {
-	tokens, err := tokenizer.CalToken(textContent)
+	r, err := tokenizer.Encode(textContent)
 	if err != nil {
 		return "", err
 	}
-	if tokens > limits {
-		return string([]rune(textContent)[:int(math.Min(float64(limits), float64(len([]rune(textContent)))))]), nil
+
+	if len(r.Bpe) > limits {
```

Comment:
How about we early return `textContent, nil` when `len(r.Bpe) <= limits`?

Reply:
updated.